ThirdPartyAdapters/vungle/vungle/src/main/java/com/vungle/mediation/VungleBannerAdapter.java

```diff
@@ -0,0 +1,265 @@
+        }
+    }
+
+    void updateVisibility(boolean visible) {
+        this.mVisibility = visible;
+        if (mVungleBannerAd != null) {
+            mVungleBannerAd.setAdVisibility(visible);
+        } else if (mVungleNativeAd != null) {
+            mVungleNativeAd.setAdVisibility(visible);
+        }
+    }
+
+    private LoadAdCallback mAdLoadCallback = new LoadAdCallback() {
+        @Override
+        public void onAdLoad(String id) {
+            createBanner();
+        }
+
+        @Override
+        public void onError(String id, VungleException exception) {
+            Log.d(TAG, "Ad load failed:" + VungleBannerAdapter.this);
+            mIsLoading.set(false);
+            mVungleManager.removeActiveBannerAd(mPlacementId);
+            if (mPendingRequestBanner && mVungleListener != null) {
+                mVungleListener.onAdFailedToLoad(AdRequest.ERROR_CODE_NO_FILL);
+            }
+        }
+    };
+
+    private PlayAdCallback mAdPlayCallback = new PlayAdCallback() {
+        @Override
+        public void onAdStart(String placementId) {
+            if (mPendingRequestBanner && mVungleListener != null) {
+                mVungleListener.onAdStart(placementId);
+            }
+        }
+
+        @Override
+        public void onAdEnd(String placementId, boolean completed, boolean isCTAClicked) {
+            if (mPendingRequestBanner && mVungleListener != null) {
+                mVungleListener.onAdEnd(placementId, completed, isCTAClicked);
+            }
+        }
+
+        @Override
+        public void onError(String placementId, VungleException exception) {
+            Log.d(TAG, "Ad play failed:" + VungleBannerAdapter.this);
+            mIsLoading.set(false);
+            mVungleManager.removeActiveBannerAd(mPlacementId);
+            if (mPendingRequestBanner && mVungleListener != null) {
+                mVungleListener.onAdFail(placementId);
+            }
+        }
+    };
+
+    private void loadBanner() {
+        if (mVungleManager.isAdPlayable(mPlacementId, mAdConfig.getAdSize())) {
+            createBanner();
+        } else if (mVungleManager.isValidPlacement(mPlacementId)) {
+            if (AdConfig.AdSize.isBannerAdSize(mAdConfig.getAdSize())) {
+                Banners.loadBanner(mPlacementId, mAdConfig.getAdSize(), mAdLoadCallback);
+            } else {
+                Vungle.loadAd(mPlacementId, mAdLoadCallback);
+            }
+        } else { // passed Placement Id is not what Vungle's SDK gets back after init/config
+            mIsLoading.set(false);
+            mVungleManager.removeActiveBannerAd(mPlacementId);
+            if (mVungleListener != null) {
+                mVungleListener.onAdFailedToLoad(AdRequest.ERROR_CODE_INVALID_REQUEST);
+            }
+        }
+    }
+
+    private void createBanner() {
+        if (!mPendingRequestBanner) {
+            mIsLoading.set(false);
+            mVungleManager.removeActiveBannerAd(mPlacementId);
```

Comment:
@HuFangshuai where this logic came from? It's different from https://github.com/Vungle/googleads-mobile-android-mediation/pull/27/files#diff-e3eb809fec019433eb4442f5941b9e9fL345

Reply:
exists in history. Seems adView is in refresh progress and pubs destroy this adView, then we don't need to create banner ad.

Reply:
Could you point me in the history where we have cleaned pending one? Thanks

Reply:
Oh..sorry, I understand your means. 
`mIsLoading.set(false);`
`mVungleManager.removeActiveBannerAd(mPlacementId);`
I added these code parts try to finish adapter instance when mPendingRequestBanner==false.
mPendingRequestBanner change to false only when destroy fired. Seems duplicate code here.